views/app.es

```diff
@@ -69,8 +138,78 @@ class Poi extends Component {
     })
   }
 
+  onLayoutChange = (layout, layouts) => {
+    if (!isLayoutsEqual(layouts, config.get('poi.mainpanel.layout'))) {
+      config.set('poi.mainpanel.layout', layouts)
+    }
+  }
+
   render() {
-    const { isHorizontal, reversed, theme } = this.props
+    const { isHorizontal, reversed, isolateGameWindow, grid, theme } = this.props
+    if (grid) {
+      return (
+        <ThemeProvider theme={theme === 'dark' ? darkTheme : lightTheme}>
+          <>
+            {config.get(
+              'poi.appearance.customtitlebar',
+              process.platform === 'win32' || process.platform === 'linux',
+            ) && (
+              <title-bar>
+                <TitleBarWrapper />
+              </title-bar>
+            )}
+            <ResizeSensor onResize={this.handleResize}>
+              <poi-main
+                style={{
+                  flexFlow: `${isHorizontal ? 'row' : 'column'}${
+                    reversed ? '-reverse' : ''
+                  } nowrap`,
+                  ...(!isHorizontal && { overflow: 'hidden' }),
+                }}
+              >
+                <ResponsiveReactGridLayout
+                  layouts={this.props.layouts}
+                  // onLayoutChange={this.onLayoutChange}
+                  rowHeight={10}
+                  margin={[3, 3]}
+                  cols={{ lg: 40, sm: 10 }}
+                  breakpoints={{ lg: 750, sm: 0 }}
+                  width={1800}
+                  isResizable={true}
+                  isDraggable={true}
+                  compactType="horizontal"
+                >
+                  <div className="teitoku-panel" key="teitoku-panel">
+                    <AdmiralPanel editable={this.props.editable} />
+                  </div>
+                  <div className="resource-panel" key="resource-panel">
+                    <ResourcePanel editable={this.props.editable} />
+                  </div>
+                  <div className="repair-panel panel-col" key="repair-panel">
+                    <RepairPanel editable={this.props.editable} />
+                  </div>
+                  <div className="construction-panel panel-col" key="construction-panel">
+                    <ConstructionPanel editable={this.props.editable} />
+                  </div>
+                  <div className="expedition-panel" key="expedition-panel">
+                    <ExpeditionPanel editable={this.props.editable} />
+                  </div>
+                  <div className="task-panel" key="task-panel">
+                    <TaskPanel editable={this.props.editable} />
+                  </div>
+                  <div className="kan-game-wrapper" key="poi-main">
+                    <KanGameWrapper key="frame" />
+                  </div>
+                </ResponsiveReactGridLayout>
+                <PoiApp />
```

Comment:
@KochiyaOcean There are 2 approach I am trying, the first one is leaving the poi app outside the grid, which resulted in the control panel taking over the whole screen and I can't figure out how to resize it.

The other approach is put this as a div like other components, I personally prefer doing so because it would then allow moving the control panel around. but if I do that, somehow the control will completely disappear

Reply:
<img width="1728" alt="Screenshot 2024-01-03 at 9 23 52 PM" src="https://github.com/poooi/poi/assets/41968256/ceb635d0-246a-43f7-9edf-a7d39c8d6e16">

when the poi app is wrapped in the grid                  
<div className="poi-app" key="miniship">
  <PoiApp />
</div>

Reply:
I think it may caused by `PoiApp` component is actually a fragment and some event bundling failed to work. Maybe you can check the console to find if there is any js wise errors. In the other way, if the DOM compoents are mounted correctly, it would be more likely to be a css issue

Reply:
I managed to get devtools to show up again but couldn't figure out what event bundling is broken, there are many error messages and don't seem to be relevant to my untrained eye. any theory based on file change in this PR?